### ISSUE TITLE
Secure connection failed for Waterfox user-agents

### ISSUE THREAD

**Issue by SpiritQuaddicted** [mholt/caddy#2073]
There are reports about people using the Waterfox fork of Firefox not being able to access certain websites. They are getting "Secure connection failed" errors. They can change their user-agents to work around this. It seems not to happen for Linux users.

https://github.com/MrAlex94/Waterfox/issues/178 mentions
- https://www.codedog.pl which has "Server: Caddy" in its headers
- https://www.beamdog.com which has "Server: Caddy" in its headers
- https://pricehipster.com which according to this fairly obscure website has had a "Caddy-Sponsors" header once (now behind Buttflare) http://pricehipster.com.rankank.com/

That's pretty much the only hint at what be going on, they are all served by Caddy.

I suspect the MITM detection to be the culprit as that is the only place where *fox mentioned in the code. Could someone take a look at this? Thanks! :)

**Comment by iaretard** [comment]
Just found https://quakewiki.org has the same issue and of course https://www.quaddicted.com.

**Comment by mholt** [comment]
What's in the server logs when the connection fails?

**Comment by mholt** [comment]
All those sites work fine for me in Waterfox. I'd love to see what the logs are.

**Comment by SpiritQuaddicted** [comment]
For https://www.quaddicted.com/ :

I installed Waterfox 56.0.4.1 (64-bit) and the addon UAControl.
I set the user-agent for quaddicted.com to "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0.2 Waterfox/56.0.4"
I get

> Secure Connection Failed
> 
> The connection to www.quaddicted.com was interrupted while the page was loading.

Clicking the cert info thing in the address bar shows the normal cert dialog and all is fine there.

I see absolutely nothing about that request in the server logs that are configured as

```
log / /var/log/caddy/access.log "{combined}"
errors /var/log/caddy/errors.log
```

With `ngrep -q "*"` all I see is one packet from my client and one packet back from the server. HTTPS so they are encrypted and I have no clue what they say...

If I set the user agent to "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0.2" (same but without the Waterfox suffix), it works fine.

If I set the user agent to "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0.2 " (trailing space), it breaks in the same way as before.
